specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/Product.json

```diff
@@ -143,6 +143,147 @@
+              "$ref": "#/definitions/ProductList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "AzureStack.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/GetProduct": {
+      "post": {
+        "x-ms-examples": {
+          "Returns the specified product.": {
+            "$ref": "examples/Product/GetPost.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns the specified product.",
+        "operationId": "GetProduct",
+        "parameters": [
+          {
+            "$ref": "AzureStack.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/RegistrationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProductNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/DeviceConfigurationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Product"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "AzureStack.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/uploadProductLog": {
+      "post": {
+        "x-ms-examples": {
+          "Returns the specified product.": {
+            "$ref": "examples/Product/UploadLog.json"
+          }
+        },
+        "tags": [
+          "Product"
+        ],
+        "description": "Returns the specified product.",
+        "operationId": "Uploadproductlog",
+        "parameters": [
+          {
+            "$ref": "AzureStack.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/RegistrationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProductNameParameter"
+          },
+          {
+            "$ref": "AzureStack.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/MarketplaceProductLogUpdate"
```

Comment:
This logs one action that was made on product
We want to use it to have a more detailed latest status for each product. It's designed on AzureStack side to send update on every action, and not batch them (just in case something goes wrong). So no reason to transform it into array.

Also, if we need to aggregate them - we can aggregate directly, without sending arrays